### PR TITLE
Assignment 3: 100% coverage

### PR DESCRIPTION
**Pull request by ipoddubny**
Ok, it should work by now.
However, I don't like the idea of mocking the Version module, it looks ugly and may break tests in parallel mode. Is there a better way to do it?

### CODE REVIEW

**Comment by EvanMarkHopkins** [comment]
I think it's a reasonable solution given the constraints. We could configure additional plugins by passing init an options parameter, but only port is specified.

File: test/version.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+var describe = lab.experiment;
+var it = lab.test;
+var expect = Code.expect;
+
+
+describe('Version.register()', function () {
+
+    it('registers /version route handler that returns version', function (done) {
+
+        var server = new Hapi.Server();
```
Comment (EvanMarkHopkins) [inline test/version.js]: I think we can safely use the init for testing version, but you could look into creating a new http server and passing the failing plugin, that would a pretty decent solution in my opinion

File: test/version.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        server.register(Version, function (err) {
+
+            expect(err).to.not.exist();
+
+            server.inject('/version', function (res) {
+
+                expect(res.statusCode).to.equal(200);
+                expect(res.result).to.contain('version');
+                expect(res.result.version).to.equal('0.0.2');
```
Comment (EvanMarkHopkins) [inline test/version.js]: Let's use the actual package.version for comparison, we shouldn't have to update this test every time the version changes. Everything else looks good :+1:

File: package.json [hunk 3]
```diff
@@ -22,7 +22,11 @@
     "hoek": "2.x.x"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "make test-cov",
```
Comment (AdriVanHoudt) [inline package.json]: this breaks for windows users

File: test/version.js (hunk 2 above)
Comment (AdriVanHoudt) [inline test/version.js]: maybe also test for the exact result. If the endpoint returns an object with an extra key the test will never catch it

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    it('returns a started hapi server', function (done) {
+
+        Index.init(null, function (err, server) {
+
+            expect(err).to.not.exist();
+
+            expect(server.info.started).to.be.above(0);
+
+            server.stop(done);
```
Comment (hussainanjar) [inline test/index.js]: Why space after each line ?

**Pull request closed without merging**